adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/AbstractVertxBasedCoapAdapter.java

```diff
@@ -715,4 +794,348 @@ protected void preShutdown() {
             final Device authenticatedDevice) {
         return Future.succeededFuture(true);
     }
+
+    /**
+     * Gets the value of the {@link org.eclipse.hono.util.Constants#HEADER_TIME_TILL_DISCONNECT} query parameter of the
+     * CoAP request.
+     *
+     * @param context The CoAP context containing the request.
+     * @return The time till disconnect or {@code null} if
+     *         <ul>
+     *         <li>the request doesn't contain a {@link org.eclipse.hono.util.Constants#HEADER_TIME_TILL_DISCONNECT}
+     *         query parameter.</li>
+     *         <li>the contained value cannot be parsed as an Integer</li>
+     *         </ul>
+     * @throws NullPointerException if context is {@code null}.
+     */
+    protected Integer getTimeUntilDisconnectFromRequest(final CoapContext context) {
```

Comment:
how about moving this method to `CoapContext`?

Reply:
Possible.

Reply:
Done.